server/src/testFixtures/java/io/crate/testing/SQLExecutor.java

```diff
@@ -774,56 +819,8 @@ public SQLExecutor addTable(String createTableStmt) throws IOException {
     /**
         * Add a table to the clusterState
         */
-    @SuppressWarnings("unchecked")
     public SQLExecutor addTable(String createTableStmt, Settings settings) throws IOException {
-        CreateTable<Expression> stmt = (CreateTable<Expression>) SqlParser.createStatement(createTableStmt);
-        CoordinatorTxnCtx txnCtx = new CoordinatorTxnCtx(CoordinatorSessionSettings.systemDefaults());
-        AnalyzedCreateTable analyzedCreateTable = createTableAnalyzer.analyze(
-            stmt, ParamTypeHints.EMPTY, txnCtx);
-        var boundCreateTable = analyzedCreateTable.bind(
-            new NumberOfShards(clusterService),
-            fulltextAnalyzerResolver,
-            nodeCtx,
-            txnCtx,
-            Row.EMPTY,
-            SubQueryResults.EMPTY
-        );
-        if (boundCreateTable.isPartitioned()) {
-            throw new IllegalArgumentException("use addPartitionedTable(..) to add partitioned tables");
-        }
-
-        var combinedSettings = Settings.builder()
-            .put(boundCreateTable.settings())
-            .put(settings)
-            .build();
-
-        ClusterState prevState = clusterService.state();
-
-        // addTable can be called multiple times, create supplier based on existing state.
-        Metadata.Builder mdBuilder = Metadata.builder(prevState.metadata());
-        LongSupplier columnOidSupplier =
-                this.columnOidSupplier != null ? this.columnOidSupplier : mdBuilder.columnOidSupplier();
-
-        RelationName relationName = boundCreateTable.tableName();
-        IndexMetadata indexMetadata = getIndexMetadata(
-            relationName.indexNameOrAlias(),
-            combinedSettings,
-            TestingHelpers.toMapping(columnOidSupplier, boundCreateTable),
-            prevState.nodes().getSmallestNonClientNodeVersion()
-        ).build();
-
-        ClusterState state = ClusterState.builder(prevState)
-            .blocks(
-                ClusterBlocks.builder()
-                    .blocks(prevState.blocks())
-                    .updateBlocks(indexMetadata)
-            )
-            .metadata(mdBuilder.put(indexMetadata, true))
-            .routingTable(RoutingTable.builder(prevState.routingTable()).addAsNew(indexMetadata).build())
-            .build();
-
-        ClusterServiceUtils.setState(clusterService, allocationService.reroute(state, "assign shards"));
-        return this;
+        return addPartitionedTable(createTableStmt, settings);
```

Comment:
What about moving all of `addPartitionedTable` to this method and get rid of `addPartitionedTable`?

Reply:
done